lib/game.rb

```diff
@@ -34,14 +34,66 @@ def message_player_display
   end
   # ============================================
 
-  def fire(declarer, coordinate)
-    if declarer == @user
-      # nested if for user error for firing on coordinate that has been fired on
-      @auto.board.cells[coordinate].fire_upon
-      # require "pry"; binding.pry
-    elsif declarer == @auto
+  def start
+    main_menu
+    players_setup_ships
+    until winner != nil
+    # results of winner
+    #fire
     end
+    end_game
+  end
+
+  def main_menu
+    print "Welcome to BATTLESHIP Enter p to play. Enter q to quit. > "
+
+    response = gets.chomp.downcase[0]
+
+    if response == "q"
+      print "Good Bye!"
+      leave_game
+    elsif response == "p"
+      print "Let's play!"
+    else response != "p" || "q"
+      print "Invalid entry, please enter p to play or q to quit. > "
+      response2 = gets.chomp.downcase[0]
+      if response2 == "p"
+        print "Let's play!"
+      elsif response2 == "q"
+        print "Good- bye!"
+        leave_game
+      else response2 != "p" || "q"
+        print "Good- bye!"
+        leave_game
+      end
+    end
+
+  end
+
+  def leave_game
+    exit
+  end
+
+  def players_setup_ships
+    @auto.ship_setup
+    @user.ship_setup
+  end
+
+
+
+  def winner
+    if @auto.ships.sum {|ship| ship.health } == 0
+      @user
+    elsif @user.ships.sum { |ship| ship.health } == 0
+      @auto
+    else
+      nil
+    end
+  end
 
+  def end_game
```

Comment:
Woo! There is the connection back to main menu! Love it!

Reply:
I'm out here! #Phew! #Glad it worked!

Reply:
<3